pkg/domain/infra/abi/trust.go

```diff
@@ -0,0 +1,171 @@
+package abi
+
+import (
+	"context"
+	"encoding/json"
+	"io/ioutil"
+	"os"
+	"strings"
+
+	"github.com/containers/libpod/pkg/domain/entities"
+	"github.com/containers/libpod/pkg/trust"
+	"github.com/pkg/errors"
+	"github.com/sirupsen/logrus"
+)
+
```

Comment:
My internal brain parses had a hard time eating the code.

```suggestion
// TODO - the trust code below has (too?) many branches and lacks comments.
// While comments are needed for making the code more approachable, a code
// simplification may be worth to make it easier to maintain in the future.
```

Reply:
this reflects the comment from above which isnt from me, i did a straight port ... im inclined to leave it.